packages/@sanity/form-builder/src/inputs/arrays/ArrayOfObjectsInput/item/ItemForm.tsx

```diff
@@ -0,0 +1,76 @@
+import {isReferenceSchemaType, Marker, Path, SchemaType} from '@sanity/types'
+import React, {ForwardedRef, forwardRef, useMemo} from 'react'
+import {FormFieldPresence} from '@sanity/base/presence'
+import {FormBuilderInput} from '../../../../FormBuilderInput'
+import {ArrayMember, ReferenceItemComponentType} from '../types'
+import PatchEvent from '../../../../PatchEvent'
+import {Props as InputProps} from '../../../types'
+
+type Props = {
+  type: SchemaType
+  value: ArrayMember
+  compareValue?: ArrayMember[]
+  markers: Marker[]
+  onChange: (event: PatchEvent) => void
+  onFocus: (path: Path) => void
+  ReferenceItemComponent: ReferenceItemComponentType
+  onBlur: () => void
+  filterField: () => any
+  isSortable: boolean
+  readOnly: boolean | null
+  focusPath: Path
+  presence: FormFieldPresence[]
+}
+
+export const ItemForm = forwardRef(function ItemForm(props: Props, ref: ForwardedRef<any>) {
+  const {
+    type,
+    value,
+    markers,
+    focusPath,
+    onFocus,
+    onBlur,
+    onChange,
+    ReferenceItemComponent,
+    readOnly,
+    isSortable,
+    filterField,
+    presence,
+    compareValue,
+  } = props
+
+  const isReference = isReferenceSchemaType(type)
+
+  const Input = useMemo(
+    () =>
+      isReference
+        ? forwardRef(function Input_(
+            givenProps: InputProps,
+            inputRef: ForwardedRef<{focus: () => void}>
+          ) {
+            return <ReferenceItemComponent {...givenProps} isSortable={isSortable} ref={inputRef} />
+          })
+        : undefined,
+    [ReferenceItemComponent, isReference, isSortable]
+  )
```

Comment:
how would you feel about inlining this logic into `ArrayItem`? i'm thinking it could simplify and remove the needed declaring all the props for this component. i haven't tried it thought so unsure of the result